src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  activeTodo: Todo;
+  setActiveTodo: (currentTodo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ activeTodo, setActiveTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [userLoading, setUserLoading] = useState(false);
+
+  useEffect(() => {
+    if (activeTodo.id) {
+      setUserLoading(true);
+    }
+
+    getUser(activeTodo.userId)
+      .then(setUser)
+      .finally(() => setUserLoading(false));
```

Comment:
and what if the request fails? 
please make sure to handle error case

Reply:
error added